plugins/Diagnostics/Diagnostic/Diagnostic.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+/**
+ * Performs a diagnostic on the system or Piwik.
```

Comment:
>  If we don't want them on the system check page then maybe they shouldn't be diagnostics indeed (e.g. super-heavy diagnostic, big output, etc..).

This is what I'm referring to, tools w/ lots of output that may take a long time to run. Obviously, they would be implemented as standalone commands.

Anyway, the issue isn't w/ the code, just referring to them as 'checks' rather than 'diagnostics' in documentation. So even if it is really needed (and if other core devs consider nomenclature to be important), we can always do it later.

On a related note: maybe it would be good to mention that Diagnostic implementations shouldn't create a lot of output or take a long time to run? That those would be better as commands?